include/node.hpp

```diff
@@ -557,20 +610,45 @@ class merged_node : public node<merged_node<Left, Right, OutId, InId>, meta::con
     }
 
     template<meta::any_simd... Ts>
-        requires meta::vectorizable_v<return_type> && input_port_types::template
-    are_equal<typename std::remove_cvref_t<Ts>::value_type...> && traits::node::can_process_simd<Left>
-            && traits::node::can_process_simd<Right> constexpr stdx::rebind_simd_t<return_type, meta::first_type<meta::typelist<std::remove_cvref_t<Ts>...>>>
-              process_one(Ts... inputs) {
-        return apply_right<InId, traits::node::input_port_types<Right>::size() - InId - 1>(std::tie(inputs...), apply_left<traits::node::input_port_types<Left>::size()>(std::tie(inputs...)));
+        requires traits::node::can_process_simd<Left> && traits::node::can_process_simd<Right>
+    [[gnu::always_inline]] constexpr meta::simdize<return_type, meta::simdize_size_v<std::tuple<Ts...>>>
+    process_one(const Ts &...inputs) {
+        static_assert(traits::node::output_port_types<Left>::size == 1,
+                      "TODO: SIMD for multiple output ports not implemented yet");
+        return apply_right<InId, traits::node::input_port_types<Right>::size() - InId - 1>(
+                std::tie(inputs...), apply_left<traits::node::input_port_types<Left>::size()>(std::tie(inputs...)));
+    }
+
+    [[gnu::always_inline]] constexpr auto
+    process_one_simd(auto N)
+        requires traits::node::can_process_simd<Right>
+    {
+        if constexpr (requires(Left &l) {
+                          { l.process_one_simd(N) };
+                      }) {
+            return right.process_one(left.process_one_simd(N));
+        } else {
+            using LeftResult = typename traits::node::return_type<Left>;
+            using V = meta::simdize<LeftResult, N>;
+            alignas(stdx::memory_alignment_v<V>) LeftResult tmp[V::size()];
+            for (std::size_t i = 0; i < V::size(); ++i) {
+                tmp[i] = left.process_one();
+            }
+            return right.process_one(V(tmp, stdx::vector_aligned));
+        }
     }
 
     template<typename... Ts>
     // Nicer error messages for the following would be good, but not at the expense of breaking
     // can_process_simd.
         requires(input_port_types::template are_equal<std::remove_cvref_t<Ts>...>)
-    constexpr return_type
+    [[gnu::always_inline]] constexpr return_type
     process_one(Ts &&...inputs) {
-        if constexpr (traits::node::output_port_types<Left>::size == 1) { // only the result from the right node needs to be returned
+        // if (sizeof...(Ts) == 0) we could call `return process_one_simd(meta::ic<width>)`. But if
```

Comment:
just code comment -> insert ic definition... maybe even refactor this as a static constexpr constant?

Reply:
Sorry, can you rephrase? I don't understand.